app/tables.py

```diff
@@ -1,11 +1,24 @@
+import os
+
+# from pathlib import Path
+
 from piccolo.apps.user.tables import BaseUser
 from piccolo.engine.sqlite import SQLiteEngine
 from piccolo_api.mfa.authenticator.tables import (
     AuthenticatorSecret as AuthenticatorSecret_,
 )
 from piccolo_api.session_auth.tables import SessionsBase
 
-DB = SQLiteEngine()
+# use this if we want the data as a directory and VOLUME /app/data
+# try:
+#     Path(Path(__file__).parent / "data").mkdir()
+# except FileExistsError:
+#     pass
+# DB_PATH = Path("data").absolute() / "auth.sqlite"
+# DB = SQLiteEngine(str(DB_PATH))
+
+DB_PATH = os.path.join(os.path.dirname(__file__), "data")
```

Comment:
Just two questions. Should the `data` be a `folder `(`directory`) with a sqlite db file inside or just a sqlite file named `data` as it is now (now is file). I think the second one should be the case (with mount `VOLUME /data` in `Dockerfile`), but I'm not sure if I understood correctly? And should I add the `data` file to `.gitignore`?

Reply:
I have to say I'm pretty confused. Your latest suggestion works `locally` and with `docker run`, but not with `docker-compose`. Should we remove `docker-compose.yml`? 
This works with all three option
```python
# in tables.py
try:
    # create data directory in root
    Path(Path(__file__).parent.parent / "data").mkdir()
except FileExistsError:
    pass
# add auth.sqlite database to data directory
DB_PATH = Path("data").absolute() / "auth.sqlite"
DB = SQLiteEngine(str(DB_PATH))
```
and mount volume in `Dockerfile`
```
VOLUME /app/data
```
Do you have any thoughts on this? In which direction should we go?

Reply:
Yes, lets put the SQLite database file in a folder called `data`.

So the folder structure will be:

```
data/
    auth.sqlite
app/
    main.py
    ...
```

That means:

- When we add the app folder to the container, there is no risk of us accidentally adding a local SQlite database
- It will work locally, and in the Docker container

As for Docker compose not working, the problem could be this:

```
    volumes:
      - .:/app
```

It might be adding entire local folder to the container. It probably means we end up with `app/app` or something in the container.

Reply:
I think it will be easier to understand so I just pushed a commit that works with all three options so you can see or try it. The directory structure is as in your last comment.